src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,140 @@
+                                                          token-requirement-list))
+
+(defn token-requirement-list [props]
+  (let [{:keys [gate]} props
+        [gate-type token-requirement-list] gate
+        is-sufficient? (is-token-requirement-met? token-requirement-list)
+        message (if
+                 (= gate-type :join)
+                  (str "Join requirements" (when-not is-sufficient? " not") " met")
+                  (if (= gate-type :read)
+                    (str "View only requirements" (when-not is-sufficient? " not") " met")
+                    (str "View and post requirements" (when-not is-sufficient? " not") " met")))]
+    [rn/view
+     [rn/view {:style (get styles :token-requirement-list-spacing)}
+      [text/text {:weight :medium
+                  :number-of-lines 1} message]
+      [text/text {:size :paragraph-2
+                  :number-of-lines 1} "You must hodl:"]]
+
+     [rn/view {:style (merge (get styles :token-row) (merge (get styles :token-requirement-list-spacing)) (get styles :token-list))}
+      (map-indexed (fn [token-index token]
+                     ^{:key token}
+                     [:<>
+                      [rn/view {:style (when-not (= token-index 0) (get styles :token-tag-spacing))}
+                       [token-tag/token-tag {:token (:token token)
+                                             :value (:amount token)
+                                             :size :big
+                                             :is-required (:is-sufficient token)
+                                             :is-purchasable (:is-purchasable token)
+                                             :token-img-src
+                                             (js/require
+                                              (str
+                                               "../resources/images/tokens/mainnet/"
+                                               (:token token)
+                                               ".png"))}]]])
+                   token-requirement-list)]]))
+
+(defn is-community-locked? [community]
+  (let [{gates :gates} community]
+    (when (some? gates)
+      (when (vector? (:join gates))
+        (not (is-token-requirement-met? (:read gates)))))))
+
+(defn is-channel-locked? [channel]
+  (let [{gates :gates} channel]
+    (when (some? gates)
+      (if (vector? (:read gates))
+        (not (is-token-requirement-met? (:read gates)))
+        (when (vector? (:write gates))
+          (not (is-token-requirement-met? (:write gates))))))))
+
+(defn token-gating [props]
+  (let [{:keys [community channel]} props
+        type (if (some? community) :community :channel)
+        inner-props (if (= type :community) community channel)
+        name (:name inner-props)
+        emoji (:emoji inner-props)
+        emoji-background-color (:emoji-background-color inner-props)
+        community-avatar (:community-avatar inner-props)
+        gates (:gates inner-props)
+        lock-status (if
```

Comment:
I updated channel avatar component recently and it now uses `locked?` prop instead of `lock-status`. Can you rebase and update accordingly? :D